Simplified/Book/UI/NYPLBookCellDelegate.m

```diff
@@ -413,33 +425,50 @@ - (void)audiobookManagerDidRequestRefresh {
   [[NYPLBookRegistry sharedRegistry] setState:NYPLBookStateDownloadNeeded forIdentifier:self.book.identifier];
 
 #if FEATURE_OVERDRIVE
-  [[NSNotificationCenter defaultCenter] addObserver:self selector:@selector(updateODAudiobookManifest) name:NSNotification.NYPLMyBooksDownloadCenterDidChange object:nil];
+  [[NSNotificationCenter defaultCenter] addObserver:self
+                                           selector:@selector(updateODAudiobookManifest)
+                                               name:NSNotification.NYPLMyBooksDownloadCenterDidChange
+                                             object:nil];
 #endif
   [[NYPLMyBooksDownloadCenter sharedDownloadCenter] startDownloadForBook:self.book];
 }
 
 #if FEATURE_OVERDRIVE
 - (void)updateODAudiobookManifest {
-  if ([[NYPLBookRegistry sharedRegistry] stateForIdentifier:self.book.identifier] == NYPLBookStateDownloadSuccessful) {
+  NYPLBookState bookState = [[NYPLBookRegistry sharedRegistry]
+                             stateForIdentifier:self.book.identifier];
+
+  if (bookState == NYPLBookStateDownloadFailed) {
```

Comment:
I added this check because it seemed like if we failed the download, it seems like the observer is not going to be removed. I could be wrong though, and maybe we do need to keep it?

Reply:
Yeah this check seems necessary